mne/io/base.py

```diff
@@ -149,19 +164,22 @@ def to_data_frame(self, picks=None, index=None, scaling_time=1e3,
                 data = data.T
                 col_names = [self.ch_names[k] for k in picks]
 
-            types = [channel_type(self.info, idx) for idx in picks]
+            ch_types = [channel_type(self.info, idx) for idx in picks]
+            ch_map = dict(
+                zip([self.info['ch_names'][pp] for pp in picks],
+                    ch_types))
             n_channel_types = 0
             ch_types_used = []
 
             scalings = _handle_default('scalings', scalings)
             for t in scalings.keys():
-                if t in types:
+                if t in ch_types:
                     n_channel_types += 1
                     ch_types_used.append(t)
```

Comment:
I know you didn't do this, but isn't this just:

```
ch_types_used = [t for t in ch_types if t in scalings]
n_channel_types = len(ch_types_used)
```

Reply:
No it is not the same. I simplified the code tough to make it more compact.